contracts/okp4-triplestore/src/msg.rs

```diff
@@ -0,0 +1,182 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Binary, Uint128};
+
+/// Instantiate message
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// Limitations regarding store usage.
+    pub limits: StoreLimits,
+}
+
+/// Execute messages
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// # Insert
+    /// Insert the Tuples extracted from the provided RDF graph.
+    /// For already existing triples it act as no-op.
+    ///
+    /// Only the smart contract owner (i.e. the address who instantiated it) is authorized to perform
+    /// this action.
+    Insert { input: GraphInput },
+
+    /// # Remove
+    /// Remove all the Tuples linked to the resources matching the criteria defined in the provided
+    /// queries.
+    ///
+    /// Only the smart contract owner (i.e. the address who instantiated it) is authorized to perform
+    /// this action.
+    Remove {
+        /// The queries act as the logical disjunction of each single query, a resource shall match
+        /// at least one query.
+        queries: Vec<ResourceQuery>,
+    },
+}
+
+/// Query messages
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// # Resources
+    ///
+    /// Returns the resources matching the criteria defined in the provided queries formatted according
+    /// to the provided format.
+    #[returns(ResourcesResponse)]
+    Resources {
+        /// The queries act as the logical disjunction of each single query, a resource shall match
+        /// at least one query.
+        queries: Vec<ResourceQuery>,
+
+        /// The expected output format. Its value shape the way the response shall be interpreted.
+        format: ResourcesOutputFormat,
+    },
+}
+
+/// # StoreLimits
+/// Contains limitations regarding store usages.
+#[cw_serde]
+pub struct StoreLimits {
+    /// The maximum number of triples the store can contains.
+    /// If `None`, there is no limit on the number of triples.
+    pub max_triple_count: Option<Uint128>,
+}
+
+/// # GraphInput
+/// Represents an RDF Graph as input supporting multiple formats.
+#[cw_serde]
+pub enum GraphInput {
+    /// Input in [RDF/XML](https://www.w3.org/TR/rdf-syntax-grammar/) format.
+    XML(Binary),
+
+    /// Input in [Turtle](https://www.w3.org/TR/turtle/) format with support of the [Turtle star](https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html#turtle-star) syntax.
+    Turtle(Binary),
+
+    /// Input in [N-Triples](https://www.w3.org/TR/n-triples/) format with support of the [N-Triples star](https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html#n-triples-star) syntax.
```

Comment:
Are you sur for the support of [N-Triples star](https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html#n-triples-star)?